lib/utils/search.js

```diff
@@ -32,12 +38,47 @@ const getWeight = (searchTerm, words, resource, resourceIndex) => {
   return weight;
 };
 
+const weightByCategories = (ids, items) => {
+  return JSON.parse(JSON.stringify(items)).map(item => {
+    item.resources = item.resources.map(resource => {
+      resource.weight = getCategoryWeight(ids, items, resource);
+      return resource;
+    });
+    return item;
+  });
+};
+
+const getCategoryWeight = (ids, categorisedResources, resource) => {
+  const categoriesResourceIsIn = categorisedResources
+    .filter(category => {
+      return category.resources.some(r => r.id == resource.id);
+    })
+    .map(x => x.id);
+
+  const selectedCategoriesResourceIsIn = ids.filter(id => {
+    return categoriesResourceIsIn.includes(id);
+  });
+
+  const unselectedCategoriesResourceIsIn = categoriesResourceIsIn.filter(id => {
+    return !ids.includes(id);
+  });
+
+  let percentageMatch = (selectedCategoriesResourceIsIn.length / ids.length) * 100;
+
+  unselectedCategoriesResourceIsIn.forEach(e => {
+    percentageMatch = percentageMatch * 0.8;
+  });
```

Comment:
What do we use `unselectedCategoriesResourceIsIn` for? Doesn't the `percentageMatch` already account for unselected categories?

I mean, when you break the Ln66 down, ...
let's say:
```
M = numMatchingCategories
N = numNotMatchingCategories
A = numAllCategories
P = percentageMatch
```
So:
```
P = M / A * 100%, where A = M +N. 
So upon substituting for All matches 'A', you get:
P = M / (M + N) * 100%
```
So, the question is: Why do we need to weigh-down the % of matching categories further, when it's already done implicitly by non-matching categories count being part of the sum in the denominator?